### ISSUE TITLE
ThreadedServer init method

### ISSUE THREAD

**Issue by skarpik** [chris-piekarski/python-json-socket#3]
Hello,

Re: jsocket_base (v1.0.1)

In the init() method of the ThreadedServer class, an instance of the JsonServer is created with the statement:

jsocket_base.JsonServer.**init**(self)

I believe that it may be better to instantiate this class with:

jsocket_base.JsonServer.**init**(self,**kwargs)

where kwargs contains the address and port. This allows one to use other than the default values:

def **init**(self, address='127.0.0.1', port=5489):

I don't usually program in Python (I prefer C++ and C#), so I may be off-base on this (the class syntax in Python is making my head hurt).

Best regards,

Steve Karpik

**Comment by chris-piekarski** [comment]
Hi Steve,

Thanks for taking the time to leave a comment. I believe I committed the change you're recommending a few years ago:
https://github.com/chris-piekarski/python-json-socket/commit/b5413c9f6b0820c1136f197040553d5bee33231c

The pypi repository does yet contain a version 1.6 and therefor wont have this commit. I will find some time to upload a new package. In the meantime you should be able to build it yourself locally.

Chris

**Comment by AnthonyARM** [comment]
@chris-piekarski Doesn't this change break the init method ? Now if I do `super(MyFactoryThread,self).__init__(address='')`
I get:
```
Apr 20 09:29:15 jenkins-blg-compute device_manager_server.py[11467]: Traceback (most recent call last):
Apr 20 09:29:15 jenkins-blg-compute device_manager_server.py[11467]:   File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
Apr 20 09:29:15 jenkins-blg-compute device_manager_server.py[11467]:     self.run()
Apr 20 09:29:15 jenkins-blg-compute device_manager_server.py[11467]:   File "build/bdist.linux-x86_64/egg/jsocket/tserver.py", line 133, in run
Apr 20 09:29:15 jenkins-blg-compute device_manager_server.py[11467]:     tmp = self._thread_type()
Apr 20 09:29:15 jenkins-blg-compute device_manager_server.py[11467]:   File "/home/jenkins/ecosystem-jenkins/scripts/device_manager_server.py", line 248, in __init__
Apr 20 09:29:15 jenkins-blg-compute device_manager_server.py[11467]:     super(MyFactoryThread,self).__init__(address='')
Apr 20 09:29:15 jenkins-blg-compute device_manager_server.py[11467]:   File "build/bdist.linux-x86_64/egg/jsocket/tserver.py", line 95, in __init__
Apr 20 09:29:15 jenkins-blg-compute device_manager_server.py[11467]:     threading.Thread.__init__(self, **kwargs)
Apr 20 09:29:15 jenkins-blg-compute device_manager_server.py[11467]: TypeError: __init__() got an unexpected keyword argument 'address'
```
By the way have you tried to have a server and client on two different machines ? I don't seem to be able to make it work

Thanks,

Anthony

**Comment by chris-piekarski** [comment]
Hi @AnthonyARM,

To overload address and port it should be done from the jsocket.ServerFactory and jsocket.JsonClient class instantiation. I just pushed a change to example_servers.py that show it working with a different port.

Yes, I have used it on different machines. I haven't confirmed its still working in a few years now. I will find some time to confirm it still works. In the meantime if you share any Traceback errors of it failing for you I'll take a look and report back.

Thanks,
Chris
